cli/mender-artifact/modify_existing_test.go

```diff
@@ -252,42 +252,46 @@ func TestModifySigned(t *testing.T) {
 	err = ioutil.WriteFile(filepath.Join(tmp, "ecdsa.key"), []byte(PrivateECDSAKey), 0711)
 	assert.NoError(t, err)
 
+	// Create and sign artifact using RSA private key.
 	os.Args = []string{"mender-artifact", "write", "rootfs-image", "-t", "my-device",
 		"-n", "release-1", "-u", filepath.Join(tmp, "mender_test.img"),
 		"-o", filepath.Join(tmp, "artifact.mender"),
 		"-k", filepath.Join(tmp, "rsa.key")}
 	err = run()
 	assert.NoError(t, err)
 
+	// Modify the artifact and re-sign it using the same RSA private key.
 	os.Args = []string{"mender-artifact", "modify",
 		"-n", "release-2",
+		"-k", filepath.Join(tmp, "rsa.key"),
 		filepath.Join(tmp, "artifact.mender")}
 
 	err = run()
-	assert.Error(t, err)
-	assert.Contains(t, err.Error(), ErrInvalidSignature.Error())
-
-	err = copyFile("mender_test.img", filepath.Join(tmp, "mender_test.img"))
 	assert.NoError(t, err)
 
+	// Modify the artifact again, this time leaving it unsigned (no private key provided).
 	os.Args = []string{"mender-artifact", "modify",
-		"-n", "release-2",
-		"-k", filepath.Join(tmp, "rsa.key"),
+		"-n", "release-3",
 		filepath.Join(tmp, "artifact.mender")}
 
 	err = run()
 	assert.NoError(t, err)
 
+	// There is a surprising side-effect (probably a bug) in the modify command:
+	// it deletes the mender_test.img in the same directory.
+	// So for now, we have to copy it again before re-creating the artifact.
```

Comment:
That is worrying indeed, I actually think this might be a showstopper for this PR until it is fixed. Just so we don't have to merge code tests that are dependent upon a bug.

Reply:
I think that this might be related to a fix I did a few weeks back, where the update would change names when modifying an artifact. Thus when the functions clean up what they think are tmp-files returned from the unpacked artifact it is really deleting the original update, as they now share the same name.
https://github.com/mendersoftware/mender-artifact/commit/e28583185d184fecf110720ad62ef32721aee0aa#diff-0f63f0dd855c7e3c955b7fa7fe2a4577

Reply:
This seems to do the trick. https://github.com/mendersoftware/mender-artifact/pull/148

Reply:
If you notice, I actually didn't change the test logic, I just added comments that explained the existing bug workaround, because it was confusing.  But yes, it would be nicer if your #148 could be merged first, then I can remove the bug workaround in #147. 

Also I created an issue for the bug:
https://tracker.mender.io/browse/MEN-2171

Reply:
Gladly! I will do that later today. Thanks, guys...

Reply:
This PR should be all ready to merge now.